architecture-decisions/0010-resource-autocomplete.md

```diff
@@ -0,0 +1,48 @@
+# 10. Resource Auto Completion
+
+Date: 2023-04-18
+
+## Status
+
+Accepted
+
+## Context
+
+Our users often don't have time to go back to a resource they've ingested and
+check to see if it's okay after the initial ingest, and so tend to immediately
+mark a resource complete at ingest time. Unfortunately this results in a lot of
+churn in our preservation infrastructure - every file that gets added during
+that ingest forces a re-preservation of the parent's metadata.
+
+To prevent that, and also make sure that resources are being completed that are
+useful to our patrons, we want to add a new state that will automatically mark a
+resource complete when all of its files are attached and the derivatives
+generated.
+
+This auto-completion could be handled in one of two ways - either via a cron
+job that runs on some regular interval or by utilizing [Sidekiq
+Batches](https://github.com/sidekiq/sidekiq/wiki/Batches).
+
+The cron job is not as immediate, but is easy to maintain and reason through.
+The batches allow us to track the lifecycle of the object through jobs and will
```

Comment:
suggestion: "would enable" instead of "will enable" -- since we're not going to do it that way